src/uu/tail/src/chunks.rs

```diff
@@ -81,3 +96,604 @@ impl<'a> Iterator for ReverseChunks<'a> {
         Some(buf[0..(block_size as usize)].to_vec())
     }
 }
+
+/// The type of the backing buffer of [`BytesChunk`] and [`LinesChunk`] which can hold
+/// [`BUFFER_SIZE`] elements at max.
+type ChunkBuffer = [u8; BUFFER_SIZE];
+
+/// A [`BytesChunk`] storing a fixed size number of bytes in a buffer.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct BytesChunk {
+    /// The [`ChunkBuffer`], an array storing the bytes, for example filled by
+    /// [`BytesChunk::fill`]
+    buffer: ChunkBuffer,
+
+    /// Stores the number of bytes, this buffer holds. This is not equal to buffer.len(), since the
+    /// [`BytesChunk`] may store less bytes than the internal buffer can hold. In addition
+    /// [`BytesChunk`] may be reused, what makes it necessary to track the number of stored bytes.
+    /// The choice of usize is sufficient here, since the number of bytes max value is
+    /// [`BUFFER_SIZE`], which is a usize.
+    bytes: usize,
+}
+
+// Silence a clippy warning about a missing default trait implementation, because a user might
```

Comment:
I think this can be solved with `derive(Default)`. And if not, we might be better off `allow`ing the warning on the struct.

Reply:
I allowed the warning